internal/services/redhatopenshift/redhat_openshift_cluster_resource.go

```diff
@@ -560,15 +584,28 @@ func flattenOpenShiftClusterProfile(profile *openshiftclusters.ClusterProfile, c
 		fipsEnabled = *profile.FipsValidatedModules == openshiftclusters.FipsValidatedModulesEnabled
 	}
 
-	return []ClusterProfile{
+	resourceGroupId, err := commonids.ParseResourceGroupIDInsensitively(*profile.ResourceGroupId)
```

Comment:
Is the API returning the resource group in the wrong casing or can we use the case sensitive parser here 
```suggestion
	resourceGroupId, err := commonids.ParseResourceGroupID(*profile.ResourceGroupId)
```